lib/charty/vector_adapters/arrow_adapter.rb

```diff
@@ -0,0 +1,130 @@
+      end
+
+      def size
+        @data.length
+      end
+
+      def where(mask)
+        mask = check_mask_vector(mask)
+        unless mask.data.is_a?(Arrow::BooleanArray)
+          mask = mask.to_a
+          mask = mask.collect {|integer| integer != 0 } if mask[0].is_a?(Integer)
+          mask = Arrow::BooleanArray.new(mask)
+        end
+        masked_data = @data.filter(mask)
+        masked_index = []
+        mask.to_a.each_with_index do |boolean, i|
+          masked_index << index[i] if boolean
+        end
+        Charty::Vector.new(masked_data, index: masked_index, name: name)
+      end
+
+      def boolean?
+        case @data
+        when Arrow::BooleanArray
+          true
+        when Arrow::ChunkedArray
+          @data.value_data_type.is_a?(Arrow::BooleanDataType)
+        else
+          false
+        end
+      end
+
+      def numeric?
+        case @data
+        when Arrow::NumericArray
+          true
+        when Arrow::ChunkedArray
+          @data.value_data_type.is_a?(Arrow::NumericDataType)
+        else
+          false
+        end
+      end
+
+      def categorical?
+        case @data
+        when Arrow::DictionaryArray
+          true
+        when Arrow::ChunkedArray
+          @data.value_data_type.is_a?(Arrow::DictionaryDataType)
+        else
+          false
+        end
+      end
+
+      def categories
+        @data.dictionary.to_a
+      end
+
+      def unique_values
+        @data.unique.to_a
+      end
+
+      def group_by(grouper)
+        grouper = Vector.new(grouper) unless grouper.is_a?(Vector)
+        group_keys = grouper.unique_values
+        grouper_data = grouper.data
+        unless grouper_data.is_a?(Arrow::Array)
+          grouper_data = Arrow::Array.new(grouper.to_a)
+        end
+        equal = Arrow::Function.find("equal")
+        group_keys.map { |key|
+          mask = equal.execute([grouper_data, key]).value
+          [key, Charty::Vector.new(@data.filter(mask))]
+        }.to_h
+      end
+
+      def drop_na
+        if @data.n_nulls.zero?
+          Vector.new(@data, index: index, name: name)
+        else
+          data_without_null =
+            Arrow::Function.find("drop_null").execute([@data]).value
+          Vector.new(data_without_null)
+        end
+      end
+
+      def eq(val)
+        mask = Arrow::Function.find("equal").execute([@data, val]).value
+        Vector.new(mask, index: index, name: name)
+      end
+
+      def notnull
+        if @data.n_nulls.zero?
+          mask = Arrow::BooleanArray.new([true] * @data.length)
+        else
+          mask = Arrow::BooleanArray.new(@data.length,
+                                         @data.null_bitmap,
+                                         nil,
+                                         0)
+        end
+        Vector.new(mask, index: index, name: name)
+      end
+
+      def mean
+        @data.mean
+      end
+
+      def stdev(population: false)
```

Comment:
Could you handle `population`?

Reply:
We need to add `arrow::compute::VariantOptions` support to red-arrow first.
I'll do it.